diagnostic.md

```diff
@@ -6,70 +6,87 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+1.
+#!/bin/bash
+API="${API_ORIGIN:-http://localhost:4741}"
+URL_PATH="/books/4"
+curl "${API}${URL_PATH}" \
+  --include \
+  --request GET
+echo
+
+2.
+Using curl in the terminal
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+1. scripts/books/show.sh
+2. Terminal
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+Runs migrations for the current environment that have not run yet.
+By default it will run migrations only in the development environment.
+It will automaticlly create a Schema.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+rails g controller things will only generate a controller for a table.
+scaffold will generate a set of MVC (model, view, controller) for a table.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+It provides a way of creating custom JSON by representing each resource as a class that inherits from the activeModel serializer.
 ```
 
 What do we expect the command `Patient.all` to return?
 
 ```md
-// your response here
+The patient's attributes
 ```
 
 We use the constant `Patient` in the `PatientsController`, but where is it
 actually defined?
 
 ```md
-// your response here
+models/patient.rb
 ```
 
 Why do our `model` files appear to be lacking so much code, when they're doing
 so much for us?
 
 ```md
-// your response here
+They don't need much to work? What we've been doing is the basics and have been generated by commands on the terminal so it doesn't need much.
 ```
 
 What does typing `bin/rails dbconsole` in a directory that houses a rails app do for
 us?
 
 ```md
-// your response here
+It figures out which database a developer is using and puts her into the command line interface she would use with it.
 ```
 
 What does typing `rails db` in a directory that houses a rails app do for us?
 
 ```md
-// your response here
+shows us the available options such as:
```

Comment:
It opens up a PostgreSQL shell connected to the database that we're working with.